nautobot/utilities/tests/test_filters.py

```diff
@@ -843,9 +844,23 @@ class MySiteFilterSet2(SiteFilterSet):
         params = {"q": "1234"}
         self.assertEqual(self.get_filterset_count(params, MySiteFilterSet2), 0)
 
+    def test_typed_icontains(self):
+        """Test a preprocessor to strip icontains (which wouldn't be by default)."""
+
+        class MySiteFilterSet(SiteFilterSet):
+            """Overload the default just to illustrate that it's all we're testing for here."""
+
+            q = SearchFilter(filter_predicates={"name": {"lookup_expr": "icontains", "preprocessor": str.strip}})
```

Comment:
Another way to do this would be leaving the default but setting `SearchFilter(strip=True), e.g.: 

```python
            q = SearchFilter(filter_predicates={"name": "icontains"}, strip=True)
```

But then that would strip all char field values across the board and not just for the `name` field, such as if there were others specified (`slug`, etc.).

Reply:
I don't think we need to add another feature to this. You're testing that providing a preprocessor works as intended